src/modules/TodoListArray.js

```diff
@@ -0,0 +1,53 @@
+class TodoListArray {
+  constructor() {
+    this.todos = JSON.parse(localStorage.getItem('todos')) || [];
+  }
+
+  addTodo(todo) {
+    this.todos.push(todo);
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  updateTodo(idx, newTodo) {
+    this.todos[idx - 1] = newTodo;
+    for (let i = 0; i < this.todos.length; i += 1) {
+      this.todos[i].idx = i + 1;
+    }
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  deleteTodo(index) {
+    if (this.todos.length === 1) {
+      this.todos = [];
+    } else {
+      this.todos.splice(index, 1);
```

Comment:
- [ ]  I would suggest using array methods that don't mutate the array such as `filter`.

   eg: 
    ```javascript 
    this.todos = this.todos.filter(todo => {todo.id != index}) ```

for more information about mutable and immutable array methods please follow this link about [How to Write Remarkably Clean JavaScript Code](https://www.sitepoint.com/immutable-array-methods-javascript/)